### DISCUSSION TITLE
Custom SSH monitoring connection timeout

### DISCUSSION THREAD

**Discussion by xibei-paolu** [apache/hertzbeat#1706] [Q&A]
Error reported when creating custom SSH monitoring: Test connection failed Peer SSH connection failed: Default AuthFuture [ssh connection]: Failed to get operation result within specified timeout: 10000
Where can I set the maximum link duration for SSH?
Custom template configuration: Customize the monitoring template YML file according to the SSH protocol configured in the development document.

**Comment by tomsun28** [comment]
hi, you can custom the timeout here. 

<img width="1237" alt="image" src="https://github.com/dromara/hertzbeat/assets/24788200/38e399bb-a870-40f4-a0bd-4e8d9126ca51">

if you use the custom monitoring template. makr sure the ssh protocol has timeout variable 


<img width="624" alt="image" src="https://github.com/dromara/hertzbeat/assets/24788200/278086f2-ac6b-4aa7-b8cb-662fded6b58c">
